Created the blog post announcing Data Prepper 2.0

dlvenable opened the pull request:
### Description

We are releasing Data Prepper 2.0.0 on Oct 10. This is our announcement blog post.

This requires the bio for @oeyh as supplied in #1067.
 
### Issues Resolved

N/A

### Check List
- [x] Commits are signed per the DCO using --signoff
- [x] Update authors list after the bio is available from @oeyh.
- [ ] Merge in #1067 


By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

dlvenable commented:
I added Hai to the list of authors based on the username he supplied in #1067. We will need to merge that in prior to this PR.

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,151 @@
+It performs host verification by checking the fingerprint of the certificate. And finally it configures each server to authenticate requests using 
+Mutual TLS (mTLS) to prevent tampering of data.
+
+## Directory structure
```
Suggested rewrites for directory structure section:

--- 

Before the release of Data Prepper 2.0, we distributed Data Prepper as a single executable JAR file. While convenient, it made it difficult for us to include custom plugins.

We now distribute Data Prepper 2.0 in a bundled directory structure. This structure features a shell script to launch Data Prepper and dedicated subdirectories for JAR files, configurations, pipelines, logs, and more.

```
data-prepper-2.0.0/
  bin/
    data-prepper                    # Shell script to run Data Prepper
  config/
    data-prepper-config.yaml        # The Data Prepper configuration file
    log4j.properties                # Logging configuration
  pipelines/                             # New directory for pipelines
    trace-analytics.yaml
    log-ingest.yaml
  lib/
    data-prepper-core.jar
    ... any other jar files
  logs/
```
You can launch Data Prepper by running `bin/data-prepper`; no need for additional command line arguments or Java system property definitions. Instead, the application loads from the `config/` subdirectory.

Data Prepper 2.0 reads pipeline configurations from the `pipelines/` subdirectory. You can now define pipelines across multiple YAML files in the subdirectory, where each file contains the settings for one or more pipelines. The directory also helps keep pipeline definition distinct and, therefore, more compact and focused.

dlvenable replied:
I added "You can _now_ launch Data Prepper..." to clarify this is new.

Also, "application loads _configurations_ from"

And I used definition here: "contains the _definition_ for one or more pipelines." I think settings is not quite clear.

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,151 @@
+which write their access logs as TSV files. Now you can parse these logs using Data Prepper. Additionally, if your events have 
+CSV or TSV fields, Data Prepper has a `csv` processor which can create fields from your incoming CSV data.
+
+## Other improvements
```
Sections edits:

---


Data Prepper 2.0 includes several other improvements. We want to highlight a few of them.

* The OpenSearch sink now supports any `create` actions for OpenSearch when writing or updating documents. Pipeline authors can configure their pipelines to create new documents when one does not already exist.
* The HTTP source now supports loading TLS/SSL credentials from either Amazon S3 or Amazon Certificate Manager. The OTel Trace Source supported these options; pipeline authors can now configure them for their log ingestion use cases.
* Data Prepper now requires Java 11, and the Docker image deploys with JDK 17.

Please see our release notes for a complete list.

Naarcha-AWS reviewed (approved):
Added my rewrites for each section in the review. One comment = one section. 

Might need to wait to add documentation links until this PR is merged: https://github.com/opensearch-project/documentation-website/pull/1510

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,151 @@
+  - technical-post
+---
+
+Today the maintainers are announcing the release of Data Prepper 2.0. It has been over a year since Data Prepper 1.0 was first introduced
```
Or maybe:

The Data Prepper maintainers are proud to announce the release of Data Prepper 2.0. This release makes Data Prepper easier to use and helps you improve your observability stack based on feedback from you, our users.

Here are some of the major changes and enhancements made for Data Prepper 2.0.

Naarcha-AWS replied:
@dlvenable: Could we add a line in this intro or somewhere in the blog about OpenSearch compatibility? Data Prepper 2.0 is compatible with all OpenSearch versions, correct?

dlvenable replied:
I added the following:

> Data Prepper 2.0 retains compatibility with all current versions of OpenSearch.

dlvenable commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,151 @@
+multiple YAML files in the subdirectory, where each file contains the configuration for one or more pipelines. This will 
+allow users to keep their pipeline definitions distinct and thus more compact and focused. 
+
+## JSON & CSV parsing
```
> This change also allows you to import CSV or TSV formatted files from Amazon S3 sources, which is helpful for systems like Amazon CloudFront, which write their access logs as TSV files.

The word "which" was repeated so I modified this somewhat.

dlvenable commented:
Thanks @Naarcha-AWS ! I took most of the changes to all sections except the Directory Structure. I want to check with @oeyh on those first.

I did make some tweaks from your suggestions - most of them were to try to be more accurate.

I also wasn't quite sure about some of the paragraphs. Did you intend all those paragraphs? The ones in the examples read too broken up and didn't keep the same train of thought.

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+
+Now Data Prepper 2.0 supports conditional routing to help pipeline authors send different logs to specific OpenSearch clusters.
+
+One common use-case this supports is to reducing the volume of data going to some clusters.
```
"Use-case" should be "use case," per our style guidelines. Change to the following:
```suggestion
One common use case for conditional routing is reducing the volume of data going to some clusters.
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+Now Data Prepper 2.0 supports conditional routing to help pipeline authors send different logs to specific OpenSearch clusters.
+
+One common use-case this supports is to reducing the volume of data going to some clusters.
+When you want info logs that produce large volumes of data to go to a cluster or index with more frequent rollovers or 
```
```suggestion
When you want info logs that produce large volumes of data to go to a cluster, index with more frequent rollovers, or add deletions to clear out large volumes of data, you can now configure pipelines to route the data with your chosen action.
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+
+
+Simply pick a name appropriate for the domain and a Data Prepper expression.  
+Then for any sink that should only have some data coming through, define one or more routes to apply Data Prepper will evaluate 
```
```suggestion
Then for any sink that should only have some data coming through, define one or more routes to apply. Data Prepper will evaluate 
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+
+Simply pick a name appropriate for the domain and a Data Prepper expression.  
+Then for any sink that should only have some data coming through, define one or more routes to apply Data Prepper will evaluate 
+these expressions for each event to determine which sinks to route these events to. Any sink that has no routes defined will accept all events.
```
```suggestion
these expressions for each event to determine which sinks to route these events. Any sink that has no routes defined will accept all events.
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+accepts log data from external sources such as Fluent Bit. 
+
+The pipeline then uses the `grok` processor to split the log line into multiple fields. 
+The `grok` processor adds named `loglevel` to the event. Pipeline authors can use that field in routes. This pipeline has two OpenSearch sinks. The first sink only receives 
```
Let's break this up a little more:

---

The pipeline then uses the `grok` processor to split the log line into multiple fields. The `grok` processor adds a named `loglevel` to the event. Pipeline authors can use that field in routes.

This pipeline contains two OpenSearch sinks. The first sink will only receive logs with a log level of `WARN` or `ERROR`. Data Prepper will route all events to the second sink.

dlvenable replied:
I took your suggestion and made one clarification by adding "field" which you can see here: "... adds a 
_field_ named `loglevel` ..."

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
````diff
@@ -0,0 +1,166 @@
+        index: all-logs
+```
+
+There are many other use-cases that conditional routing can support. If there are other conditional expressions 
````
```suggestion
There are many other use cases that conditional routing can support. If there are other conditional expressions 
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+
+Data Prepper 2.0 introduces peer forwarding as a core feature.
+
+Previous to Data Prepper 2.0, performing stateful trace aggregations required using the peer-forwarder processor plugin. 
```
Is there a distinction between "peer-forwarder" and peer forwarder? Should this be:

```suggestion
Previous to Data Prepper 2.0, performing stateful trace aggregations required using the peer forwarder processor plugin. 
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+
+## JSON & CSV parsing
+
+Many of our users have incoming data with embedded JSON or CSV fields. To help in these use-cases, Data Prepper 2.0 
```
```suggestion
Many of our users have incoming data with embedded JSON or CSV fields. To help in these use cases, Data Prepper 2.0 
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+which write their access logs as TSV files. Now you can parse these logs using Data Prepper. 
+
+Additionally, if your events have 
+CSV or TSV fields, Data Prepper 2.0 now contains a `csv` processor which can create fields from your incoming CSV data.
```
To reduce the number of times, we use "which," we can change it in this instance to "that". This might get changed in again in editorial, so while I'm leaving my suggestion I'll defer to @natebower on how best to avoid repetition here.

```suggestion
CSV or TSV fields, Data Prepper 2.0 now contains a `csv` processor that can create fields from your incoming CSV data.
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+Data Prepper 2.0 includes a number of other improvements. We want to highlight a few of them.
+
+* The OpenSearch sink now supports `create` actions for OpenSearch when writing documents. Pipeline authors can configure their pipelines to only create new documents and not update existing ones.
+* The HTTP source now supports loading TLS/SSL credentials from either Amazon S3 or Amazon Certificate Manager. The OTel Trace Source supported these options; pipeline authors can now configure them for their log ingestion use-cases.
```
```suggestion
* The HTTP source now supports loading TLS/SSL credentials from either Amazon S3 or Amazon Certificate Manager. Pipeline authors can now configure them for their log ingestion use cases. Before Data Prepper 2.0, only the OTel Trace Source supported these options.
```

Naarcha-AWS commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+
+* The OpenSearch sink now supports `create` actions for OpenSearch when writing documents. Pipeline authors can configure their pipelines to only create new documents and not update existing ones.
+* The HTTP source now supports loading TLS/SSL credentials from either Amazon S3 or Amazon Certificate Manager. The OTel Trace Source supported these options; pipeline authors can now configure them for their log ingestion use-cases.
+* Data Prepper now requires Java 11 or higher, and the Docker image deploys with JDK 17.
```
```suggestion
* Data Prepper now requires Java 11 or higher. The Docker image deploys with JDK 17.
```

Naarcha-AWS reviewed (changes requested):
A few more minor tweaks before we pass them off to @natebower.

dlvenable commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,166 @@
+deletions to clear out these large volumes of data, you now configure pipelines to route your data.
+
+
+Simply pick a name appropriate for the domain and a Data Prepper expression.  
+Then for any sink that should only have some data coming through, define one or more routes to apply Data Prepper will evaluate 
```
Line breaks should not affect the rendered page.

dlvenable replied:
There was a space and line break which did create a new paragraph in the rendered page. Thanks for noting that!

dlvenable commented:
It took all the suggested changes.

natebower commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,168 @@
+* The HTTP source now supports loading TLS/SSL credentials from either Amazon S3 or Amazon Certificate Manager. Pipeline authors can now configure them for their log ingestion use cases. Before Data Prepper 2.0, only the OTel Trace Source supported these options.
+* Data Prepper now requires Java 11 or higher. The Docker image deploys with JDK 17.
+
+Please see our [release notes](https://github.com/opensearch-project/data-prepper/releases/tag/2.0.0) for a complete list.
```
The only thing we're missing here is a call to action. We need to conclude with a couple sentences telling the reader what we'd like for them to do next or where they can go to learn more. The below is an example from a recent blog post announcing Snapshot Management (SM):

**Wrapping it up**

SM automates taking snapshots of your cluster and provides useful features like notifications. To learn more about SM, check out the [SM documentation section](https://opensearch.org/docs/latest/opensearch/snapshots/snapshot-management/). For more technical details, read the [SM meta issue](https://github.com/opensearch-project/index-management/issues/280).

If you’re interested in snapshots, consider contributing to the next improvement we’re working on: [searchable snapshots](https://github.com/opensearch-project/OpenSearch/issues/3895).

natebower reviewed (commented):
@dlvenable Please see my changes and comments, and let me know if you have any questions. Thanks!

dlvenable commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,168 @@
+With peer forwarding as a core feature, pipeline authors can perform stateful 
+aggregations on multiple Data Prepper nodes. When performing stateful aggregations, Data Prepper uses a hash ring to determine 
+which nodes are responsible for processing different events based on the values of certain fields. Peer forwarder 
+routes events to the node responsible for processing the event. That node then holds all the state necessary for performing the aggregation.
```
I'm not sure about the change to "states" here. Using a singular noun for state is quite common.

> In information technology and computer science, a system is described as stateful if it is designed to remember preceding events or user interactions; the remembered information is called the state of the system.

https://en.wikipedia.org/wiki/State_(computer_science)

dlvenable commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,168 @@
+Data Prepper 2.0 includes a number of other improvements. We want to highlight a few of them.
+
+* The OpenSearch sink now supports `create` actions for OpenSearch when writing documents. Pipeline authors can configure their pipelines to only create new documents and not update existing ones.
+* The HTTP source now supports loading TLS/SSL credentials from either Amazon S3 or Amazon Certificate Manager. Pipeline authors can now configure them for their log ingestion use cases. Before Data Prepper 2.0, only the OTel Trace Source supported these options.
```
I believe either `SSL/TLS` or `TLS/SSL` is in use. I intentially chose `TLS/SSL` because we are using TLS. The SSL part is mostly there for historical reasons.

You can also see that the term `TLS/SSL` is used in the following Wikipedia article.

https://en.wikipedia.org/wiki/Transport_Layer_Security

dlvenable reviewed (commented):
I pushed all the changes except the final call-to-action section. I will push that soon.

dlvenable commented:
> @dlvenable I changed to "states" to match @Naarcha-AWS edits and also to avoid "all the state". If you want to use "state", remove "all".

That makes sense. I've removed "all" from the sentence.

dlvenable commented:
I have also pushed a short conclusion section.

natebower commented on _posts/2022-10-10-Announcing-Data-Prepper-2.0.0.md:
```diff
@@ -0,0 +1,173 @@
+
+## Try Data Prepper 2.0
+
+Data Prepper 2.0 is available for [download](https://opensearch.org/downloads.html#data-prepper) now. The maintainers encourage you to
```
Because this is a blog, an exclamation point would work after the first sentence. Other than that small nit, LGTM.

natebower reviewed (dismissed):
LGTM

krisfreedain reviewed (approved):
looks good!
